src/middlewares.rs

```diff
@@ -12,11 +12,11 @@ use base64::{engine::general_purpose::STANDARD_NO_PAD, Engine};
 use serde::{Deserialize, Serialize};
 use tracing::error;
 
-use crate::scraper::ThinCourse;
+use crate::scraper::{ThinCourse, ThinSection};
 
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct UserState {
-    pub selection: Vec<ThinCourse>,
+    pub selection: Vec<(ThinCourse, Vec<ThinSection>)>,
```

Comment:
I've been thinking about this, but I want to do some restructuring with cookies, if we think this is a good idea:

Instead of having one big cookie that holds the state for everything, I would prefer to have smaller cookies, with each cookie contains the selections for a single term only. This allows having an expiration date on each term's schedule, so we don't run out of storage client-side (browsers can store max ~4kb of cookie, metadata included.)

If so, I would say keep this as `Vec`, since it's a lot faster to iterate through 3-7 entries to find a selection than using `HashMap`

Reply:
I can't imagine a situation where we're trying to store 4kb of data as long as a HashMap is serializable in some sensible way. I don't think we need the added complexity of storing a separate cookie per term

As far as what the userstate struct should look like... I'm thinking this:

```rust
#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct UserState {
    pub selection: Map<Term, Map<ThinCourse, Vec<ThinSection>>>,
}
```

The maps being HashMap is an implementation detail as far as I'm aware

Reply:
I actually wouldn't mind implementing per-term userstate, that makes sense to me to fragment it a little. I could go ahead and tack that onto this PR real quick.

This way, we could do something like this:
```rust
pub struct UserState {
    pub selection: Map<Term, Set<Section>>,
}
```
And the term could be inferred from the current request. I don't see a need to ever access another term's UserState  with the current requirements.

Reply:
LGTM!

Reply:
Yeah that looks great!

Reply:
Okay, it should be doing that now. I just dropped a FAT commit 318464f